### ISSUE TITLE
Help! Adetailer results in loss of skin texture with SDXL models but not SD1.5 models.

### ISSUE THREAD

**Issue by JPGranizo** [Bing-su/adetailer#449]
### Question

I observed that using Adetailer with SDXL models (both Turbo and Non-Turbo variants) leads to an overly smooth skin texture in upscaled faces, devoid of the natural imperfections and pores.

On the other hand, the combination of Adetailer with SD1.5 models noticeably enhances the skin's details and imperfections.

This conclusion was drawn from extensive testing with multiple SDXL Turbo, SDXL Non-Turbo, and SD1.5 models. Can you please advise why there is such a difference and how to solve this issue?

Below are some comparative examples (the left side shows images without Adetailer, and the right side demonstrates the results with Adetailer)

**RealvisXL 3 Turbo:**

![test_c_2](https://github.com/Bing-su/adetailer/assets/57099674/9c7d3a3f-3a2b-4b7a-a9fd-ed4c51980c81)

![test_c_1](https://github.com/Bing-su/adetailer/assets/57099674/e6395555-f62e-4618-86c8-fe03ba613fd0)

**RealvisXL 3 Non-Turbo**

![test_c_1_nt](https://github.com/Bing-su/adetailer/assets/57099674/5280f1e7-d2fd-46ef-8fa3-0e5e7dd62f1a)

![test_c_2_nt](https://github.com/Bing-su/adetailer/assets/57099674/b2e2052d-6d58-4a14-a438-3a8e22bd4386)

**SD1.5 Cyberrealistic v41 and AnalogMadness 7**

<img width="1891" alt="SD1 5_test" src="https://github.com/Bing-su/adetailer/assets/57099674/68bff12e-84fd-409b-9f6f-9832abc47459">

<img width="1904" alt="SD1 5_test_2" src="https://github.com/Bing-su/adetailer/assets/57099674/9bbbe207-7581-400f-b645-860bf4b0ca79">

**Adetailer settings: default**
https://i.imgur.com/5ko6BX8.png

**Comment by Bing-su** [comment]
I've seen this happen before when using adetailer on images where faces make up a large percentage of the overall image.
Try disabling adetailer based on your usage.

**Comment by JPGranizo** [comment]
@Bing-su Thank you for your quick response and the insightful suggestion.

The example images I presented were specifically cropped to highlight the facial area and better illustrate the issue. However, in the original images, the faces actually comprise a small portion of the overall composition.

I'm still puzzled as to why Adetailer enhances skin details for SD1.5 models but tends to overly smooth out skin textures in SDXL models. This discrepancy becomes even more pronounced in images featuring individuals with freckles, where Adetailer, when used with SDXL models, noticeably diminishes these skin details.

Could there be an underlying reason for this distinct difference in the behavior of Adetailer between SD1.5 and SDXL/Turbo? Your expertise and insights on this matter would be greatly valued.

**Comment by JPGranizo** [comment]
@Bing-su Here is an example demonstrating a significant reduction in skin texture (the freckles) using Adetailer with RealvisXL 3.0 Turbo. Face is only occupying a small part of the composition. 

**Cropped image: left without Adetailer, right with Adetailer**

<img width="1912" alt="freckles example_sdxl turbo" src="https://github.com/Bing-su/adetailer/assets/57099674/4b01e09f-d2e5-4132-bbf3-7d919db103ef">

**Original image: left without Adetailer, right with Adetailer**

<img width="1899" alt="freckles total" src="https://github.com/Bing-su/adetailer/assets/57099674/a65a8446-60df-49ac-886e-bbdbd3446c6b">

**Comment by Toad8787** [comment]
I don't understand how that would actually be because of adetailer, which more automatically finds the item in question and then in-paints automatically, but more of how in-painting works itself.  But then again idk

**Comment by derpina-ai** [comment]
this is true, I have noticed this as well. It's a side effect of inpainting, in some cases the detail is lost. Adetailer should enforce higher CFG in most cases I believe, as this is most likely where the detail could be lost, the rest is up to denoising strength, if you use large denoising, even default 0.4, chances are your details are  gone, and probably not replaced.

**Comment by JPGranizo** [comment]
@derpina-ai for Turbo models I increase the steps fpr Adetailer. When I create the image with Turbo-merges I use 7-10 steps. For Face in-painting, I use 20 steps. That adds enough details again, so that it is not too different from the rest of the image

**Comment by Woisek** [comment]
@JPGranizo I often have original 30 steps for generation that's then used for inpainting, still the "smoothed" faces appear, with little to no details.

**Comment by JPGranizo** [comment]
> @JPGranizo I often have original 30 steps for generation that's then used for inpainting, still the "smoothed" faces appear, with little to no details.

I tried to get to the bottom of this. I tested Adetailer (A1111) vs standard inpainting (A1111) and FaceDetailer (ComfyUI). The results are similar for each method - smoothed faces. Increase the inpainting steps improves the situation slightly, but does not completely fix it.

This is an issue of XL models, as they are really bad at inpainting. I even tried specialised XL inpainting models. In my tests, they didnt perform any better regarding this issue compared to standard non-inpainting XL models.
